sql/deployments/deployment-2018-01-16-COLAB-2495.sql

```diff
@@ -0,0 +1,18 @@
+# Display all entries that have the issue
+SELECT * FROM xcolab.members_Member WHERE members_Member.status IS  NULL ;
+
+# Display entries without a createDate and the replacement createDate (DryRun)
+SELECT T1.id_, T1.screenName, T1.createDate, T2.createDate
+FROM xcolab.members_Member T1, xcolab.members_Member T2
+WHERE T1.id_ = T2.id_ + 1 AND T1.createDate IS NULL;
+
+# Update status fields
+UPDATE xcolab.members_Member SET status=0 WHERE status IS NULL;
+UPDATE xcolab.members_Member SET autoRegisteredMemberStatus=0 WHERE autoRegisteredMemberStatus IS NULL;
+
+# Update createDate field
+UPDATE xcolab.members_Member AS T1
+JOIN xcolab.members_Member AS T2
+    ON T1.id_-1 = T2.id_
+SET T1.createDate = T2.createDate
+WHERE T1.createDate IS NULL;
```

Comment:
Did this query populate all createDate fields when you ran it locally?

Two potential issues:
1. The ids range might not be contiguous everywhere.
2. If two consecutive createDates are null, the order matters. Is this dependency resolved correctly?

One potential solution could be to find the closest member with a create date in a subquery like this:
```sql
SELECT id_, ABS( id_ - myId ) AS distance
FROM members_Member
WHERE createDate is not null
ORDER BY distance
LIMIT 1
```

Reply:
The script did work for all of the rows that I encountered. So 1 and 2 were not an issue for this one. I can however improve the script to account for those issues if you want.

Reply:
If it worked for all instances, then I think it's good enough - no need to over-engineer it :)